### PR TITLE
Fixes around feature-flagging, particularly in oxql, as well as uuid-kinds, sled-storage

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Related to https://github.com/oxidecomputer/omicron/pull/6018, but without the CI/xtask bits.

### CODE REVIEW

**Review by david-crespo — APPROVED** [review]
Fixes my problem. See https://github.com/oxidecomputer/omicron/pull/6018#issuecomment-2221035718

File: oximeter/db/src/lib.rs [hunk 1]
```diff
@@ -35,11 +41,12 @@ pub mod model;
 #[cfg(any(feature = "oxql", test))]
 pub mod oxql;
 pub mod query;
-#[cfg(any(feature = "oxql", feature = "sql", test))]
+#[cfg(any(feature = "oxql", feature = "sql", feature = "oxdb", test))]
```
Comment (bnaecker) [inline oximeter/db/src/lib.rs:44]: Is this mod needed on the `oxdb` feature? It looks like you've moved the `make_client()` method, and I don't see any others called unless the `oxql` or `sql` is also enabled.

**Review by bnaecker — APPROVED** [review]
Looks good to me! One question, not a blocker. Thanks for splitting this out, makes a lot of sense while you continue to figure out the best path on CI feature-checking.

File: oximeter/db/src/lib.rs (hunk 1 above)
Comment (zeeshanlakhani) [reply oximeter/db/src/lib.rs:44]: yep, just leftover.

Comment (zeeshanlakhani) [reply oximeter/db/src/lib.rs:44]: making a few more minor updates around oxdb and this, coming up.

**Pull request merged**